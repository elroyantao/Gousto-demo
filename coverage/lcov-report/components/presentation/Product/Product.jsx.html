<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/presentation/Product/Product.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">components/presentation/Product</a> Product.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>2/2</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react'
import PropTypes from 'prop-types'
&nbsp;
import Accordion from '../Accordion/Accordion'
&nbsp;
export default class Product extends Component {
  static propTypes = {
    product: PropTypes.object.isRequired
  }
&nbsp;
  static defaultProps = {
    product: {}
  }
&nbsp;
  render() {
    const { product } = this.props
    return (
      &lt;li className="Product"&gt;
        &lt;Accordion header={product.title}&gt;
          &lt;div className="Product-body"&gt;{product.description}&lt;/div&gt;
        &lt;/Accordion&gt;
      &lt;/li&gt;
    )
  }
}
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Jul 20 2017 11:00:38 GMT+0100 (BST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
