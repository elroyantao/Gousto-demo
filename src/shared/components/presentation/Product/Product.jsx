import React, { Component } from 'react'
import PropTypes from 'prop-types'

import Accordion from '../Accordion/Accordion'

export default class Product extends Component {
  static propTypes = {
    product: PropTypes.object.isRequired
  }

  static defaultProps = {
    product: {}
  }

  render() {
    const { product } = this.props
    return (
      <li className="Product">
        <Accordion header={product.title}>
          <div className="Product-body">{product.description}</div>
        </Accordion>
      </li>
    )
  }
}
