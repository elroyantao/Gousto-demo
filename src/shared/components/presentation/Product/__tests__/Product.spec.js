import testComponentHelper from '../../../../lib/test-helper'
import Product from '../Product'

describe('<Product/>', () => {
  const renderComponent = testComponentHelper(Product)
  const initialProps = {
    product: {
      title: 'this is the title',
      description: 'this is the description'
    }
  }
  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent(initialProps).getTree())
        .toMatchSnapshot()
    })
    it('with empty product', () => {
      expect(renderComponent({}).getTree())
        .toMatchSnapshot()
    })
  })
})
