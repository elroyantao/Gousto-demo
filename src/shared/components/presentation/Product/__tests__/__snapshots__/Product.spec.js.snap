// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Product/> @renders in default state 1`] = `
<li
  className="Product"
>
  <Accordion
    header="this is the title"
  >
    <div
      className="Product-body"
    >
      this is the description
    </div>
  </Accordion>
</li>
`;

exports[`<Product/> @renders with empty product 1`] = `
<li
  className="Product"
>
  <Accordion>
    <div
      className="Product-body"
    />
  </Accordion>
</li>
`;
